pkg/kubelet/images/image_gc_manager.go

```diff
@@ -256,6 +280,16 @@ func (im *realImageGCManager) detectImages(ctx context.Context, detectTime time.
 	// Make a set of images in use by containers.
 	for _, pod := range pods {
 		for _, container := range pod.Containers {
+			// Skip containers that are not actually using the image
+			// This includes containers in error states like CreateContainerConfigError
+			if !isContainerActuallyUsingImage(container) {
```

Comment:
Does this mean that if image is not used by any container it will never be gc-ed?

Reply:
@bart0sh  No, that's not the case.

If an image is not used by any container, it is eligible for garbage collection according to the usual image GC policies (such as imageMaximumGCAge and disk pressure thresholds).

This change only ensures that images used by containers in error states (like CreateContainerConfigError) are not considered "in use" and can be garbage collected if not used elsewhere.

Let me know if you have any other questions!

Reply:
> This change only ensures that images used by containers in error states (like CreateContainerConfigError) are not considered "in use" and can be garbage collected if not used elsewhere.

How does this fix the case described in the PR description: `images used by containers in error states (such as CreateContainerConfigError) were repeatedly garbage collected and re-downloaded, causing unnecessary network traffic and resource usage.` ?
According to the change, images referenced by containers with `CreateContainerConfigError` will not be considered as `in use` and therefore will be gc-ed and re-downloaded, no?

Reply:
@aryasoni98 Please, address my above comment, thanks!